lightning/src/ln/peer_handler.rs

```diff
@@ -1637,6 +1690,24 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 
 			for (descriptor, peer_mutex) in peers.iter() {
 				self.do_attempt_write_data(&mut (*descriptor).clone(), &mut *peer_mutex.lock().unwrap());
+
+				// Only see if we have room for onion messages after we've written all channel messages, to
+				// ensure they take priority.
```

Comment:
Of course, I agree that channel messages have a higher priority than onion messages, however if we take the subset of the gossips messages, I'm not sure it's true. Reasoning, onion messages paths are likely to be correlated on payment paths, a routing hop might have an incentive to fasten the traffic of those onions as they might be odds of future routing fees. Gossips are just local host resources consumption, with only loose incentives for the hops (e.g let's say the gossip announced a channel update in this other part of the graph). Just a high level idea not sure it's relevant to introduce now that level of fine-grained class of messages priorities.